zio-http/jvm/src/main/scala/zio/http/netty/client/NettyClientDriver.scala

```diff
@@ -155,26 +150,28 @@ final case class NettyClientDriver private[netty] (
       }
     }
 
-    f.ensuring(
-      ZIO
-        .unless(location.scheme.isWebSocket) {
-          // If the channel was closed and the promises were not completed, this will lead to the request hanging so we need
-          // to listen to the close future and complete the promises
-          NettyFutureExecutor
-            .executed(channel.closeFuture())
-            .interruptible
-            .zipRight(
-              // If onComplete was already set, it means another fiber is already in the process of fulfilling the promises
-              // so we don't need to fulfill `onResponse`
-              onComplete.interrupt && onResponse.fail(
-                new PrematureChannelClosureException(
-                  "Channel closed while executing the request. This is likely caused due to a client connection misconfiguration",
+    f.ensuring {
+      // If the channel was closed and the promises were not completed, this will lead to the request hanging so we need
+      // to listen to the close future and complete the promises
+      ZIO.unless(location.scheme.isWebSocket) {
+        ZIO.succeedUnsafe { implicit u =>
+          channel.closeFuture().addListener { (_: ChannelFuture) =>
+            // If onComplete was already set, it means another fiber is already in the process of fulfilling the promises
+            // so we don't need to fulfill `onResponse`
+            nettyRuntime.unsafeRunSync {
+              ZIO.whenZIO(onComplete.interrupt)(
+                onResponse.fail(
+                  new PrematureChannelClosureException(
+                    "Channel closed while executing the request. This is likely caused due to a client connection misconfiguration",
+                  ),
```

Comment:
This change is not strictly necessary but I think adding a listener to Netty's close future is a much cleaner approach than forking an interruptible effect like previously